docs/LearnCalico.md

```diff
@@ -0,0 +1,364 @@
+to Calico networking and no security policy has been configured at this point, 
+so they do not yet have connectivity or Calico-assigned IP addresses.  
+
+The workloads do not have connectivity between one another, but they can access 
+the internet via the Docker bridge interface. None of the Calico processes have 
+performed any new actions at this point.
+
+![docker run](diagrams/docker_run.png)
+
+**Note**: Docker's libnetwork plugin works a bit differently than this 
+default-networking example. When using Calico with libnetwork, this step and the 
+next three steps in the process all occur at the same time.  The container and 
+profile are created simultaneously, the container is added to Calico networking, 
+and the profile is set on the container.  Continue reading to learn more about 
+what happens under the covers.
+
+### Add Containers to Calico Networking
+
+The workloads are added to Calico networking by calling:
+
+    calicoctl container add <container_id> <ip_address>
+
+The `<container_id>` can be either the name of the container or the container's 
+workload id.  The `<ip_address>` must be an address that falls within a 
+configured IP pool.  The default IP pools for Calico are `192.168.0.0/16` for 
+IPv4 and `fd80:24e2:f998:72d6::/64` for IPv6 (we'll use IPv4 in this example). 
+You can view pools by running `calicoctl pool show`. To add a new pool, run 
+`calicoctl pool add <cidr>` or `calicoctl pool range add <start_ip> <end_ip>`. 
+Etcd stores pools as `/calico/v1/ipam/v4/pool/192.168.0.0-16`, where assigned 
+IPs in a pool are stored as: `/calico/v1/ipam/v4/assignment/192.168.0.0-16/192.168.1.1`.
+
+When a workload is added to Calico, the `calicoctl` tool checks the etcd 
+datastore to confirm that the IP address passed in falls under a previously 
+defined IP pool. It then creates a veth pair on the host for the new endpoint, 
+storing one end in the host namespace and moving the other end into the 
+container.  After successfully creating the veth pair, `calicoctl` saves the 
+workload id, the endpoint id, the endpoint's assigned IP, the host's IP, the 
+container's new veth MAC address, and the name of the veth on the host's end 
+to the etcd datastore as:
+
+    # KEY
+    /calico/v1/host/Host1/workload/docker/<workload_id>/endpoint/<endpoint_id>
+    
+    # VALUE
+    {
+     "ipv6_gateway": null,
+     "state": "active",
+     "name": "<host_veth_name>",
+     "ipv4_gateway": "<host_ip>",
+     "ipv6_nets": [], "profile_ids": [],
+     "mac": "<container_mac>",
+     "ipv4_nets": ["<assigned_ip>"]
+    }
+
+Felix reads the changes in etcd. Felix then programs routes to the container's 
+IP address via the host's veth.  Felix also, by default, programs ACLs into the 
+host's Linux kernel ip tables that drop all traffic to the container. A security 
+profile will be used to configure inbound and outbound traffic rules later (see below).
+
+BIRD sees that new routes have been added to the host's routing table.  BIRD 
```

Comment:
Also, BIRD also adds routes to the other hosts based on the container IPs on that host.